test/integration/sync.test.js

```diff
@@ -280,7 +309,7 @@ function testFunc() {
           it('should save the entities from the backend in the cache', (done) => {
             storeToTest.pull()
               .then((result) => {
-                return validatePullOperation(result, [entity1, entity2])
+                return validatePullOperation(result, [entity1, entity2]);
```

Comment:
I prefer to keep validations in the test itself. Adding functions else where makes it more difficult to follow validations  and require the user to scroll to multiple places. DRY principles are great but I think expectations in unit tests is one area that it doesn't make sense.

Can you put all validations (expectations) inside the unit test?

Reply:
These are the integration tests.

Having long files anywhere would make you scroll - why does it make less sense in tests, than in source code? I think that's fine. If anything, keeping the assertions stable for all relevant cases is more important in tests - you must know once you've broken something. Keeping them in sync if you have them copied 20 times in a file makes it more risky than not having tests - false positives are worse than no tests as they give you confidence.

I'm inclined to agree that some repetition can be more convenient in tests, but I'd still prefer to avoid it. Like I may be OK with having 3 different suites for CacheStore, SyncStore and NetworkStore, instead of looping over them, cause they have identical CRUD API (though I still lean toward looping, slightly). But the assertions, I see no benefit of repeating the same block of code dozens of times.

In this file, I've added a function called `simulateBLPreSaveHook`. It acts only on the assertions. I made the change in two places. If we had these copied 15 times (the number of tests currently in the sync suite, and it's still not complete), I'd have to do it 15 times.

I think having them this way is fine. But either way, refactoring the integration tests code isn't part of this feature. I've merely added a test to an existing suite.

It's also worth pointing out that these same changes were done outside of the redesign, as it made more sense adding them to master at this point, rather than when we release the redesign. So there was a separate PR for that - the commit after PR can be seen [here](https://github.com/Kinvey/js-sdk/pull/214/commits), as part of the ongoing integration tests effort.

Reply:
Ok